### PR TITLE
added legend to barplot and boxplot [preprocess tab]. Added customiza…

### LINKED ISSUE #62: Add legend to plots (Pre-Process)
**What:** 
In preprocessing tab there are tabs for barplot and boxplot that are colored by design matrix. There is no legend to know what the colors mean. 

**How to replicate:** 
1. Load demo data in Load Data 
2. Move to pre-process tab 
3. Note plots 

**Extra Extra** 
Could add options for multiple choices of design matrix

### PR DESCRIPTION
**Pull request by bderenge**
…tion options to PCA. Added code for generating an rmarkdown report of the Preprocess tab. Currently the download button for the report is commented out so the option is not avaiable to users. Updated idep_Workflow to be more usable for debugging and function writing.
fixes #62

### CODE REVIEW

File: R/mod_02_pre_process.R [hunk 1]
```diff
@@ -622,6 +629,48 @@ mod_02_pre_process_server <- function(id, load_data, tab) {
...
+      # For PDF output, change this to "report.pdf"
+      filename ="report.pdf",
+      content = function(file) {
+        # Copy the report file to a temporary directory before processing it, in
+        # case we don't have write permissions to the current working dir (which
+        # can happen when deployed).
+        #tempReport <- file.path(tempdir(), "test_workflow.Rmd")
+        #tempReport
+        #tempReport<-gsub("\\", "/",tempReport,fixed = TRUE)
+        markdown_location <-"C:/Users/bdere/Documents/GitHub/idepGolem/vignettes/test_workflow2.Rmd"
```
Comment (espors) [inline R/mod_02_pre_process.R:643]: This is hard coded to your address?

Comment (espors) [reply R/mod_02_pre_process.R:643]: @bderenge

**Review by espors — CHANGES_REQUESTED** [review]
- Please add legend title (Pre-process), even if its just "Groupings" from "groupings" 
- PCA tab looks great! 
- Since the downloading of the report is not available to users yet, please remove this and put it in into a different pull request. 
- These are separate issues on separate tabs. Please consider submitting a different pull request per "fix" in the future.

File: R/mod_02_pre_process.R (hunk 1 above)
Comment (bderenge) [reply R/mod_02_pre_process.R]: I've added a commit that both fixes the hard-coding location and also comments out the report downloading section. those areas will be added back later in separate pull request.

**Review by espors — APPROVED** [review]

**Pull request merged**